jax/api.py

```diff
@@ -1967,6 +1967,61 @@ def device_put(x, device: Optional[xc.Device] = None):
   return tree_map(lambda y: xla.device_put_p.bind(y, device=device), x)
 
 
+def device_put_sharded(x: Sequence[Any], devices: Sequence[xc.Device]) -> pxla.ShardedDeviceArray:
+  """Shards the input to specified devices, returning ShardedDeviceArrays.
+
+  Args:
+    x: A sequence of arrays, scalars, or (nested) standard Python containers thereof.
+    devices: A sequence of devices()
+
+  Returns:
+    A ShardedDeviceArray or (nested) Python container thereof containing a copy
+    of x sharded across the specified devices.
```

Comment:
```suggestion
    A ShardedDeviceArray or (nested) Python container thereof containing a stacked
    version of x sharded across the specified devices.
```

I'm less attached to this suggestion, but same idea as above.